javaparser-core/src/main/java/com/github/javaparser/resolution/SymbolResolver.java

```diff
@@ -22,7 +22,21 @@
 package com.github.javaparser.resolution;
 
 import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.type.Type;
+import com.github.javaparser.resolution.types.ResolvedType;
 
 public interface SymbolResolver {
-    <T> T resolve(Node node, Class<T> resultClass);
+    /**
+     * For a reference it would find the corresponding
+     * declaration.
+     */
+    <T> T resolveDeclaration(Node node, Class<T> resultClass);
+
+    /**
+     * For types it would find the corresponding resolved types.
+     */
+    <T> T toResolvedType(Type javaparserType, Class<T> resultClass);
```

Comment:
I feel that `resolveType` means the same thing as `toResolvedType` and is more in line with `resolveDeclaration`.

Reply:
I am not very good with naming, especially in English. The difference I see is that I can have a reference (es. "foo") and from that I want to find the correspinding declaration (es. "private int foo;"). This is what `resolveDeclaration` does. `toResolvedType` instead takes a JP type and give me a JSS type. Sure, to build the JSS type it does all the necessary resolving. However to me it seems more like a conversion. Making we can calling `correspondingResolvedType` or something like that.

Reply:
I'm mostly thinking about "discoverability" - if I type a variable and a dot in my IDE and scroll through the autosuggested methods, the methods from this interface would not show up next to each other.

But that's a small thing. Keep it as it is then :-)